src/util.js

```diff
@@ -101,39 +102,69 @@ define([
         };
         /**
          * Register an event handler to be called each time an event is fired.
+         *
+         * @param type - Event type string.
+         * @param method - Event handler method.
+         * @param parameters - Parameters to be passed to method. If null
+         *      or not provided, the event object itself will be passed.
+         * @param unbindOn - (optional) Event type on which to unbind
+         *      all handlers associated with `context`. To make a one-
+         *      shot, use the same value for this parameter as for
+         *      `type`.
+         * @param context - (optional) Context for `unbind`. The
+         *      default is `null`. The handler (and all other handlers
+         *      bound to the same context) will be unbound the next time
+         *      the `unbindOn` event fires or `this.unbind(context)` is
+         *      called, whichever happens first.
          */
-        that.on = function (type, method, parameters, bindingContext) {
+        that.on = function (type, method, parameters, unbindOn, context) {
+            if (arguments.length < 5) {
+                context = null;
+            }
             var handler = {
                 method: method,
                 parameters: parameters,
-                bindingContext: bindingContext || method
+                context: context
             };
             if (registry.hasOwnProperty(type)) {
                 registry[type].push(handler);
             } else {
                 registry[type] = [handler];
             }
+            if (unbindOn) {
+                if (!unbinders[unbindOn]) {
+                    unbinders[unbindOn] = [];
+                }
+                if (unbinders[unbindOn].indexOf(context) === -1) {
+                    unbinders[unbindOn].push(context);
```

Comment:
This is so only one unbinder is created for a given context. This makes the unbinding more efficient than it was before because now many different event handlers can use the same unbinder without a globally shared context.